models/deepseek_v4_flash_dspark/utils.py

```
@@ -534,3 +556,803 @@ def quant_w_per_channel(w: torch.Tensor) -> tuple[torch.Tensor, torch.Tensor]:
+    if total_pages > CSA_MAIN_PAGES_AT_1M * max(request_count, 1):
+        raise ValueError("CSA fixture exceeds the per-request 1M page ceiling")
+
+    def allocate(
+        explicit: list[list[int]] | None,
+        pool_pages: int | None,
+        allocation_seed: int,
+    ) -> list[list[int]]:
+        if explicit is not None:
+            if len(explicit) != request_count:
+                raise ValueError("explicit CSA page ids must match valid_ranges")
+            if any(len(ids) != count for ids, count in zip(explicit, counts)):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)